code/game/objects/items/radio/headset.dm

```diff
@@ -231,9 +234,48 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 		camera.toggle_cam(null, FALSE)
 	if(sl_direction)
 		disable_sl_direction()
+	remove_minimap()
 	to_chat(wearer, "<span class='notice'>You toggle the Squad HUD off.</span>")
 	playsound(loc, 'sound/machines/click.ogg', 15, 0, 1)
 
+/obj/item/radio/headset/mainship/proc/add_minimap()
+	var/type
+	switch(frequency)
+		if(FREQ_ALPHA)
+			type = /datum/action/minimap/alpha
+		if(FREQ_BRAVO)
+			type = /datum/action/minimap/bravo
+		if(FREQ_CHARLIE)
+			type = /datum/action/minimap/charlie
+		if(FREQ_DELTA)
+			type = /datum/action/minimap/delta
+		else
+			type = /datum/action/minimap/marine
+	var/datum/action/minimap/mini = new type
+	mini.give_action(wearer)
+	if(!wearer.job.minimap_icon)
+		return
+	if(wearer.assigned_squad)
+		SSminimaps.add_marker(wearer, wearer.z, mini.minimap_flags, lowertext(wearer.assigned_squad.name)+"_"+wearer.job.minimap_icon)
+		return
+	SSminimaps.add_marker(wearer, wearer.z, mini.minimap_flags, wearer.job.minimap_icon)
```

Comment:
It means marines will only see members of their own squad? And that command can't see them either?

Reply:
Even if I'm not understanding that correctly and that marine can see command, and command can see them, I'm not sure it's really good to prevent marines from different squad to see each other. I guess you are trying to encourage squad play, but in practice I'm not sure it will really happen, at least not just thanks to minimap. 

Segregating the squads like that will just make your map less usefull, which is very sad considering it can be game changing in the way marine organise themselves

Reply:
This was bugged, I fixed it
As for  specifics I'd liek to see how it acts before making any big plans